service/src/test/java/com/google/auto/service/processor/AutoServiceProcessorTest.java

```diff
@@ -46,4 +48,30 @@ public void autoService() {
             JavaFileObjects.forResource("META-INF/services/test.SomeService"),
             JavaFileObjects.forResource("META-INF/services/test.AnotherService"));
   }
-}
\ No newline at end of file
+
+  @Test
+  public void multiService() {
+    assert_().about(javaSources())
+        .that(Arrays.asList(
+            JavaFileObjects.forResource("test/SomeService.java"),
+            JavaFileObjects.forResource("test/MultiServiceProvider.java"),
+            JavaFileObjects.forResource("test/AnotherService.java")))
+        .processedWith(new AutoServiceProcessor())
+        .compilesWithoutError()
+        .and().generatesFiles(
+            JavaFileObjects.forResource("META-INF/services/test.SomeService"),
```

Comment:
I think you'd want these to be a different file from the above test's file.

Alternatively, we can probably use `JavaFileObjects.forSourceLines()` for both so we don't need to rely on the actual paths of a checked-in-file

Reply:
problem seems to be deeper. I misunderstood the error message, and it seems the issue is that it's not finding the generated file, not the expected ones to compare them to. Wondering if it has to do with the compile-testing update

Reply:
Try using the `Compilation` and `CompilationSubject` apis? Those might be easier to use for this

Reply:
There's no API in compile-testing's clauses to get the underlying `Compilation` from what I can see. Do you have an example of what you mean?